contracts/examples/Collateralized.sol

```diff
@@ -60,158 +70,309 @@ contract Collateralized is TermsContract {
-        collateralForAgreementID[agreementID] = collateral;
-
         // emit event that collateral has been secured.
-        CollateralLocked(agreementID, token, amount);
+        CollateralLocked(agreementId, collateralToken, collateralAmount);
+
+        return true;
     }
 
+    /**
+     * Returns collateral to the debt agreement's original collateralizer
+     * if and only if the debt agreement's term has lapsed and
+     * the total expected repayment value has been repaid.
+     *
+     * @param agreementId bytes32 The debt agreement's ID
+     */
     function returnCollateral(
-        bytes32 agreementID
+        bytes32 agreementId
     )
         public
     {
-        // fetch relevant collateral instance.
-        Collateral memory collateral = collateralForAgreementID[agreementID];
+        // fetch all relevant collateralization parameters
+        address collateralToken;
+        uint collateralAmount;
+        uint gracePeriodInDays;
+
+        (collateralToken, collateralAmount, gracePeriodInDays) = retrieveCollateralParameters(agreementId);
 
-        // Ensure a valid form of collateral is tied to this agreement id.
-        require(collateral.lockupPeriod > 0);
+        // Ensure a valid form of collateral is tied to this agreement id
+        require(collateralAmount > 0);
 
-        // Collateral can only be returned after the lockup period.
-        require(block.timestamp > collateral.lockupPeriod);
+        // Withdrawal can only occur if the collateral has yet to be withdrawn.
+        // When we withdraw collateral, we reset the collateral agreement
+        // in a gas-efficient manner by resetting the address of the collateralizer to 0
+        require(agreementToCollateralizer[agreementId] != address(0));
 
-        // Collateral can only be returned if it has yet to be withdrawn.
-        require(!collateral.withdrawn);
+        // Ensure that the debt agreement's term has lapsed
+        require(getTermEndTimestamp(agreementId) < block.timestamp);
 
-        // ensure sufficient payment.
+        // Ensure that the debt is not in a state of default
         require(
-            getExpectedRepaymentValue(agreementID, block.timestamp) <=
-            getValueRepaidToDate(agreementID)
+            getExpectedRepaymentValue(
+                agreementId,
+                block.timestamp
+            ) <= getValueRepaidToDate(agreementId)
         );
 
-        // mark collateral as withdrawn.
-        collateralForAgreementID[agreementID].withdrawn = true;
+        // determine collateralizer of the collateral.
+        address collateralizer = agreementToCollateralizer[agreementId];
 
-        // transfer the collateral this contract was holding in escrow back to sender.
+        // Mark agreement's collateral as withdrawn by setting the agreement's
+        // collateralizer to 0x0.
+        delete agreementToCollateralizer[agreementId];
+
+        // seize collateral and transfer to beneficiary.
```

Comment:
the original comment `// transfer the collateral this contract was holding in escrow back to sender.` seemed more accurate

Reply:
Agreed, fixing.